lib/AST/Decl.cpp

```diff
@@ -7904,6 +7908,12 @@ void Decl::setClangNode(ClangNode Node) {
   *(ptr - 1) = Node.getOpaqueValue();
 }
 
+ErrorDecl::ErrorDecl(DeclContext *DC, SourceLoc Loc)
+    : Decl(DeclKind::Error, DC), Loc(Loc) {
+  assert(DC->getASTContext().Diags.hadAnyError() &&
```

Comment:
Let's make this unreachable so it applies even outside of asserts builds.

Reply:
You mean this?
```cpp
if (!DC->getASTContext().Diags.hadAnyError()) {
  llvm_unreachable("'ErrorDecl' should not be constructed without errors");
}
```